### DISCUSSION TITLE
Adding to premined coins

### DISCUSSION THREAD

**Discussion by ghost** [0xPolygon/polygon-edge#866] [General]
# [ Subject of the issue ]
## Description
Good morning. My premine script was off by one zero. I needed 1 billion coins, but only mined 100 million. Is there a way to mint 900 million more without recreating genesis again?

## Your environment
- OS and version. Ubuntu 20.04LTS 
- The version of the Polygon Edge.    
  (*Confirm the version of your Polygon edge client by running the following command: `polygon-edge version --grpc-address GRPC_ADDRESS`*)
- The branch that causes this issue.
- Locally or Cloud hosted (which provider).
- Please confirm if the validators are running under containerized environment (K8s, Docker, etc.).

## Steps to reproduce
- Tell us how to reproduce this issue. Put 100000000000000000000000000 in the premine field 
- Where the issue is, if you know.
- Which commands triggered the issue, if any.
- Provide us with the content of your genesis file.
- Provide us with commands that you used to start your validators.
- Provide us with the peer list of each of your validators by running the following command: `polygon-edge peers list --grpc-address GRPC_ADDRESS`.
- Is the chain producing blocks and serving customers atm?

## Expected behavior
- Tell us what should happen.
- Tell us what happened instead.

## Logs
Provide us with debug logs from all of your validators by setting logging to `debug` output with: `server --log-level debug`

## Proposed solution
If you have an idea on how to fix this issue, please write it down here, so we can begin discussing it

**Comment by laviniat1996** [comment]
Hi @Nexsuperne101 ,

Thank you for asking this. I have converted your issue to a discussion, since this is a better fit.

The premined amount of the coins will be the total supply. No other amount of the native currency (coins) can be minted later.

**Comment by ghost** [comment]
I found a quick and dirty way to this, without regenerating the secrets again etc etc. Delete the current testnet genesis.json, delete the data-dir/node/blockchain and trie folders. Run the new genesis script and then put the seal line in. Starts from block 1 and with the coins I needed. Obviously you can't do this on a live mainnet.
